package-res/resources/web/pentaho/visual/base/model.js

```diff
@@ -45,37 +73,287 @@ define([
      * @description Creates a base `Model`.
      * @param {pentaho.visual.base.spec.IModel} modelSpec A plain object containing the model specification.
      */
-    var Model = Complex.extend({
-          meta: {
-            id: "pentaho/visual/base",
-            view: "View",
-            isAbstract: true,
-            props: [
-              {
-                name: "width",
-                type: "number",
-                isRequired: true
-              },
-              {
-                name: "height",
-                type: "number",
-                isRequired: true
-              },
-              {
-                name: "isInteractive",
-                type: "boolean",
-                value: true
-              },
-              {
-                name: "data",
-                type: "object",
-                isRequired: true
-              }
-            ]
+    var Model = Complex.extend().implement(EventSource).extend(/** @lends pentaho.visual.base.Model# */{
```

Comment:
It's a temporary mechanism that allows overriding _emit to a version that swallows exceptions